samples/ef-core/json/json-columns.cs

```diff
@@ -0,0 +1,171 @@
+//It requires Oracle EF Core 8 or higher. Oracle Database 21c and higher supports JSON columns.
+//Earlier database versions map aggregate types to NCLOB columns instead of JSON columns.
+
+//Specify the user, password, and data source in the connection string below.
+
+using Microsoft.EntityFrameworkCore;
+
+namespace ODPJsonColumns
+{
+    class Program
+    {
+        public class ContactDetails
+        {
+            public Address Address { get; set; } = null!;
+            public string? Phone { get; set; }
+        }
+    
+        public class Address
+        {
+            public Address(string street, string city, string postcode, string country)
+            {
+                Street = street;
+                City = city;
+                Postcode = postcode;
+                Country = country;
+            }
+      
+            public string Street { get; set; }
+            public string City { get; set; }
+            public string Postcode { get; set; }
+            public string Country { get; set; }
+        }
+    
+        public class Author
+        {
+            public int Id { get; set; }
+            public string? Name { get; set; }
+            public ContactDetails? Contact { get; set; }
+        }
+
+        public class AuthorContext : DbContext
+        {
+            public DbSet<Author> Authors { get; set; }
+
+            //To use Oracle database JSON columns, connect to Oracle Database 21c or higher version and
+            // specify OracleSQLCompatibility to DatabaseVersion21 or higher.
+            protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+            {
+                optionsBuilder.UseOracle("User Id=<USER>; Password=<PASSWORD>; Data Source=<DATA SOURCE>"
+                    , b => b.UseOracleSQLCompatibility(OracleSQLCompatibility.DatabaseVersion21)); ;
+            }
+
+            protected override void OnModelCreating(ModelBuilder modelBuilder)
+            {
+                // Configuring the Contact column as a JSON column
+                modelBuilder.Entity<Author>().OwnsOne(
+                    author => author.Contact, ownedNavigationBuilder =>
+                    {
+                        ownedNavigationBuilder.ToJson();
+                        ownedNavigationBuilder.OwnsOne(contactDetails => contactDetails.Address);
+                    });
+            }
+        }
+
+        public static void Main()
+        {
+            using (var context = new AuthorContext())
+            {
+                context.Database.EnsureDeleted();
+                context.Database.EnsureCreated();
+          
+                var author1 = new Author()
+                {
+                    Name = "John Smith",
+                    Contact = new ContactDetails()
+                    {
+                    Phone = "555 123 4567",
+                    Address = new Address("1 Any Street", "Austin", "78741", "US")
```

Comment:
indentation

Reply:
Good catch! The same PR will fix this issue: https://github.com/oracle/dotnet-db-samples/pull/343